keras_retinanet/bin/debug.py

```diff
@@ -200,19 +203,30 @@ def run(generator, args, anchor_params):
                 # draw regressed anchors in green to override most red annotations
                 # result is that annotations without anchors are red, with anchors are green
                 draw_boxes(image, annotations['bboxes'][max_indices[positive_indices], :], (0, 255, 0))
-
+	
+            # display name on the image	
+            if args.name:	
+                draw_caption(image, [0, image.shape[0]], ntpath.basename(generator.image_path(i)))
+                
         cv2.imshow('Image', image)
         key = cv2.waitKey()
-        # note that the right and left keybindings are probably different for windows
-        # press right for next image
-        if key == 83:
+
+        ## note that the right and left keybindings are probably different for windows
+        ## press right for next image and left for previous (linux)
+        ## macOS: previous = 'n' | next = 'm'
+
+        if key == 109:
             i += 1
-        # press left for previous image
-        if key == 81:
+            if i == (generator.size() - 1): # Terminate after last image.
```

Comment:
Hmm good catch. Could you change it to the following:

- Remove the `--loop` argument.
- Change the right action to `i = (i + 1) % generator.size()`
- Change the left action to `if i < 0: i = generator.size() - 1`

That way we have looping with the left / right key.

Reply:
nice one!